### PR TITLE
Remove extra trailing newlines from preprocessor.js output

### PR DESCRIPTION
**Pull request by sbc100**
There were two different causes of extra newlines here.

### CODE REVIEW

File: src/parseTools.js [hunk 1]
```diff
@@ -61,6 +61,11 @@ function preprocess(filename) {
   const isHtml = (fileExt === 'html' || fileExt === 'htm') ? true : false;
   let inStyle = false;
   const lines = text.split('\n');
+  // text.split yeild an extra empty line at the end if text itself ends with a newline.
```
Comment (kripken) [inline src/parseTools.js:64]:
```suggestion
  // text.split yields an extra empty line at the end if text itself ends with a newline.
```

File: src/parseTools.js [hunk 2]
```diff
@@ -61,6 +61,11 @@ function preprocess(filename) {
...
+  if (text[text.length-1] == '\n') {
+    lines.pop()
```
Comment (kripken) [inline src/parseTools.js:66]:
```suggestion
    lines.pop();
```

File: src/parseTools.js [hunk 3]
```diff
@@ -61,6 +61,11 @@ function preprocess(filename) {
...
+  }
```
Comment (kripken) [inline src/parseTools.js:67]: Though, checking `text` and popping `lines` seems less natural to me than checking `lines` and popping it? That is,
```js
if (lines[lines.length - 1] == '\n') {
  lines.pop();
}
```

**Review by kripken — APPROVED** [review]
lgtm % question

File: src/parseTools.js (hunk 3 above)
Comment (sbc100) [reply src/parseTools.js:67]: Yes I admit looks odd.. I thought i had good reason to do it this way but I'm not sure now.. will go with your version,

**Pull request merged**